### PR TITLE
[#17410] migrate status-im.ethereum.core

### LINKED ISSUE #17410: migrate status-im.ethereum.core

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #17410

### CODE REVIEW

File: src/status_im/browser/core.cljs [hunk 1]
```diff
@@ -352,22 +353,29 @@
                 :id      (int id)
                 :result  result}}})
 
+(defn utf8-to-hex
+  [s]
+  (let [hex (native-module/utf8-to-hex (str s))]
+    (if (empty? hex)
+      nil
```
Comment (ibrkhalil) [inline src/status_im/browser/core.cljs:360]: Would an empty string be better?

File: src/status_im/ens/core.cljs [hunk 2]
```diff
@@ -225,6 +226,12 @@
       :searching)
     :else :invalid))
 
+(defn addresses-without-watch
+  [db]
+  (into #{}
+        (remove #(= (:type %) :watch)
+                (map #(eip55/address->checksum (:address %)) (get db :profile/wallet-accounts)))))
```
Comment (ibrkhalil) [inline src/status_im/ens/core.cljs:233]: Are we sure this won't be lazy and that it'll go through of the elements?

File: src/status_im2/subs/networks.cljs [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+            [status-im2.config :as config]
+            [utils.ethereum.chain :as chain]))
 
 (defn- filter-networks
   [network-type]
   (fn [network]
-    (let [chain-id (ethereum/network->chain-id network)
-          testnet? (ethereum/testnet? chain-id)
+    (let [chain-id (chain/network->chain-id network)
+          testnet? (chain/testnet? chain-id)
```
Comment (ibrkhalil) [inline src/status_im2/subs/networks.cljs:10]: kebab case?

**Review by ibrkhalil — APPROVED** [review]

File: src/status_im/browser/core.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/browser/core.cljs:360]: i'm just moving stuff

File: src/status_im/ens/core.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im/ens/core.cljs:233]: i'm just moving stuff

File: src/status_im2/subs/networks.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im2/subs/networks.cljs:10]: i'm just moving stuff

**Review by qfrank — APPROVED** [review]

File: src/status_im/browser/core.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/browser/core.cljs:360]: this is old code, so  we shouldn't care much

File: src/status_im/ens/core.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im/ens/core.cljs:233]: this is old code, so we shouldn't care much

File: src/status_im2/subs/networks.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im2/subs/networks.cljs:10]: what do you mean?

**Comment by flexsurfer** [comment]
@status-im/mobile-qa just need confirmation that e2e not related to this PR

**Comment by VolodLytvynenko** [comment]
Hi @flexsurfer, the failures are not caused by this PR. All of them are related to message delivery issues. 
Thank you!

**Pull request merged**